helix-term/src/ui/editor.rs

```diff
@@ -250,7 +250,9 @@ impl EditorView {
 
             // Special-case: cursor at end of the rope.
             if range.head == range.anchor && range.head == text.len_chars() {
-                spans.push((cursor_scope, range.head..range.head + 1));
+                if i != primary_idx {
```

Comment:
Since the primary index is now ignored, L245-249 can be simplified and `primary_cursor_scope` removed altogether.